src/main/java/by/itech/lab/supplier/auth/handler/CustomAuthenticationSuccessHandler.java

```diff
@@ -0,0 +1,32 @@
+package by.itech.lab.supplier.auth.handler;
+
+import by.itech.lab.supplier.auth.UserImpl;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component("customSuccessHandler")
+public class CustomAuthenticationSuccessHandler implements AuthenticationSuccessHandler {
+
+    private final ObjectMapper mapper;
+
+    public CustomAuthenticationSuccessHandler(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    @Override
+    public void onAuthenticationSuccess(HttpServletRequest request,
+                                        HttpServletResponse response, Authentication authentication)
+            throws IOException {
+        UserImpl principal = (UserImpl) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        response.getWriter().append(mapper.writeValueAsString(principal.getCustomer()));
```

Comment:
![image](https://user-images.githubusercontent.com/44436165/96725030-24ab0f80-13b9-11eb-8e6c-5a0a91153af8.png)
what about using built-in mechanisms for writing value in the response? is this not possible here?

Reply:
what do you mean? 
something like this:    .failureHandler((req, resp, ex) -> resp.setStatus(SC_FORBIDDEN))

Reply:
there is a lot of ways to do this, you need just choose one of them and use, but dont use low level stuff